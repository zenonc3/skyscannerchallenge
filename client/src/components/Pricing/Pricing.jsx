import PropTypes from 'prop-types';
import BpkButton from 'bpk-component-button';
import BpkText from 'bpk-component-text';
import React from 'react';
import STYLES from './Pricing.scss';

const c = className => STYLES[className] || 'UNKNOWN';

const Pricing = ({ price, agent, bookingUrl }) => (
  <div className={c('Pricing__container')}>
    <div className={c('Pricing__detail')}>
      <BpkText className={c('Pricing__price')} tagName="p" textStyle="lg" bold>{price}</BpkText>
      <BpkText className={c('Pricing__provider')} tagName="p" textStyle="sm">{agent}</BpkText>
    </div>
    <div className={c('Pricing__select')}>
      <BpkButton link={bookingUrl}>Select</BpkButton>
    </div>
  </div>
);

Pricing.propTypes = {
  price: PropTypes.string.isRequired,
  agent: PropTypes.string.isRequired,
  bookingUrl: PropTypes.string.isRequired,
};

export default Pricing;
