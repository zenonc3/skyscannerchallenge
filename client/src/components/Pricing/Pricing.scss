@import '~bpk-mixins';

.Pricing {
  &__container {
    display: flex;
    width: 100%;
    flex-flow: wrap row;
  }

  &__detail {
    display: flex;
    flex-direction: column;
  }

  &__select {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-end;
    flex: 1;
  }

  &__price {
    color: grayscale($color: #000);
  }

  &__provider {
    color: grayscale($color: #999);
  }
}
