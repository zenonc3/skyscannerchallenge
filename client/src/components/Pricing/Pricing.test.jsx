import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Pricing from './Pricing';

describe('Pricing', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Pricing agent="myagent" bookingUrl="http://url.com" price={30} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

