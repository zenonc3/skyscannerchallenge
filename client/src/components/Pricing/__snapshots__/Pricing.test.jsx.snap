// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing should render correctly 1`] = `
<div
  className="Pricing__container"
>
  <div
    className="Pricing__detail"
  >
    <BpkText
      bold={true}
      className="Pricing__price"
      tagName="p"
      textStyle="lg"
    >
      Â£
      30
    </BpkText>
    <BpkText
      bold={false}
      className="Pricing__provider"
      tagName="p"
      textStyle="sm"
    >
      myagent
    </BpkText>
  </div>
  <div
    className="Pricing__select"
  >
    <BpkButton
      blank={false}
      className={null}
      destructive={false}
      disabled={false}
      featured={false}
      href={null}
      iconOnly={false}
      large={false}
      link={false}
      onClick={[Function]}
      outline={false}
      rel={null}
      secondary={false}
      submit={false}
    >
      Select
    </BpkButton>
  </div>
</div>
`;
